src/interfaces/simulation.ts

```diff
@@ -136,57 +171,72 @@ export class SimulationInterface<T extends ChainId> extends ServiceInterface<T>
-      if (!options.slippage) {
-        throw new SdkError("slippage needs to be specified for a zap", SdkError.NO_SLIPPAGE);
-      }
-      let needsApproving: boolean;
 
-      if (fromVault === EthAddress) {
-        needsApproving = false;
-      } else {
-        needsApproving = await this.yearn.services.zapper
-          .zapOutApprovalState(from, fromVault)
-          .then((state) => !state.isApproved)
-          .catch(() => {
-            throw new ZapperError("zap out approval state", ZapperError.ZAP_OUT_APPROVAL_STATE);
-          });
-      }
+    const [vault] = await this.yearn.vaults.get([fromVault]);
 
-      forkId = needsApproving ? await this.simulationExecutor.createFork() : undefined;
-      options.forkId = forkId;
-      const approvalSimulationId = needsApproving
-        ? await this.yearn.services.zapper
-            .zapOutApprovalTransaction(from, fromVault, "0")
-            .catch(() => {
-              throw new ZapperError("zap out approval transaction", ZapperError.ZAP_OUT_APPROVAL);
-            })
-            .then(async ({ from, to, data }) => {
-              return this.simulationExecutor.makeSimulationRequest(from, to, data, { ...options, save: true });
-            })
-            .then((res) => res.simulation.id)
-        : undefined;
+    if (!vault) {
+      throw new SdkError(`Could not get vault: ${fromVault}`);
+    }
 
-      options.root = approvalSimulationId;
+    const token = await this.yearn.tokens.findByAddress(toToken);
 
-      simulateWithdrawal = (save: boolean): Promise<TransactionOutcome> => {
-        options.save = save;
-        return this.zapOut(from, toToken, underlyingToken, amount, fromVault, needsApproving, options);
-      };
-    } else {
-      simulateWithdrawal = (save: boolean): Promise<TransactionOutcome> => {
-        options.save = save;
-        return this.directWithdraw(from, toToken, amount, fromVault, vaultContract, options);
+    if (!token) {
+      throw new SdkError(`Could not find the token by address: ${toToken}`);
+    }
+
+    const isUnderlyingToken = await this.isUnderlyingToken({ vault: fromVault, address: token.address });
+
+    if (isUnderlyingToken) {
+      const simulateFn = (save: boolean): Promise<TransactionOutcome> => {
+        return this.directWithdraw({ ...withdrawArgs, vaultContract, options: { ...options, save } });
       };
+      return this.simulationExecutor.executeSimulationWithReSimulationOnFailure(simulateFn);
+    }
+
+    const { isZapOutSupported, zapOutWith } = getZapOutDetails({ chainId: this.chainId, token, vault });
+
+    if (isZapOutSupported && zapOutWith) {
+      return this.handleZapOutSimulation({ withdrawArgs, zapOutWith, token });
+    }
+
+    throw new SdkError(`Withdraw from ${fromVault} to ${toToken} is not supported`);
+  }
+
+  private async getWithdrawApprovalData({ from, fromVault, options }: GetWithdrawApprovalData) {
+    if (fromVault === EthAddress) {
+      return { needsApproving: false };
+    }
```

Comment:
We can remove this validation since ETH is not a valid vault, and would have already failed in past validations.

Reply:
Good catch, was using the old logic. Removed with https://github.com/yearn/yearn-sdk/pull/286/commits/52e58669a34bee474b4601147542867366fed3fb